subsys/net/lib/sockets/sockets_tls.c

```diff
@@ -3028,11 +3028,18 @@ static int ztls_socket_data_check(struct tls_context *ctx)
 		/* MbedTLS API documentation requires session to
 		 * be reset in other error cases
 		 */
-		ret = tls_mbedtls_reset(ctx);
-		if (ret != 0) {
+		int err = tls_mbedtls_reset(ctx);
+
+		if (err != 0) {
 			return -ENOMEM;
 		}
 
+#if defined(CONFIG_NET_SOCKETS_ENABLE_DTLS)
+		if (ret == MBEDTLS_ERR_SSL_TIMEOUT && ctx->type == SOCK_DGRAM) {
+			/* DTLS timeout interpreted as closing of connection. */
+			return -ENOTCONN;
```

Comment:
Maybe `ETIMEDOUT` could be used instead?

Reply:
Setting the ENOTCONN as the return value check in ztls_poll_update_pollin sets ZSOCK_POLLHUP with it.
But could add that as well, if that is more readable.

Reply:
Ah ok, in that case better to leave it as is I think.